src/ERC1155A.sol

```diff
@@ -307,37 +276,90 @@ abstract contract ERC1155A is IERC1155A {
         return true;
     }
 
-    /// @notice Internal function for decreasing single id approval amount
-    /// @dev Only to be used by address(this)
-    /// @dev Notice `owner` param, only contract functions should be able to define it
-    /// @dev Re-adapted from ERC20
-    function _decreaseAllowance(
-        address owner,
-        address spender,
+    /*///////////////////////////////////////////////////////////////
+                    SERC20 AND TRANSMUTE LOGIC 
+    //////////////////////////////////////////////////////////////*/
+
+    /// @inheritdoc IERC1155A
+    function registerSERC20(
         uint256 id,
-        uint256 subtractedValue
+        string memory name,
+        string memory symbol
     )
-        internal
+        external
```

Comment:
anyone can register an sERC20 for any id with any random name and symbol

Reply:
Made the function virtual and leaving up to the contract implementer to decide how to implement it